File: org.hl7.fhir.convertors/src/main/java/org/hl7/fhir/convertors/SpecPackage.java
package org.hl7.fhir.convertors;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

import org.hl7.fhir.model.core.StructureDefinition;
import org.hl7.fhir.model.core.ValueSet;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@AllArgsConstructor
@NoArgsConstructor
@Data
public class SpecPackage {
  private List<ValueSet> valuesets = new ArrayList<ValueSet>();
  private List<ValueSet> expansions = new ArrayList<ValueSet>();
  private Map<String, StructureDefinition> types = new HashMap<String, StructureDefinition>();
  private Map<String, StructureDefinition> resources = new HashMap<String, StructureDefinition>();
  private Map<String, StructureDefinition> extensions = new HashMap<String, StructureDefinition>();
  private Map<String, StructureDefinition> profiles = new HashMap<String, StructureDefinition>();
}

File: org.hl7.fhir.convertors/src/main/kotlin/org/hl7/fhir/convertors/SpecPackage.kt
package org.hl7.fhir.convertors

import org.hl7.fhir.r5.model.StructureDefinition
import org.hl7.fhir.r5.model.ValueSet
import java.util.HashMap

data class SpecPackage (
    var valuesets: Map<String, ValueSet> = HashMap(),
    var expansions: Map<String, ValueSet> = HashMap(),
    var types: Map<String, StructureDefinition> = HashMap(),
    var resources: Map<String, StructureDefinition> = HashMap(),
    var extensions: Map<String, StructureDefinition> = HashMap(),
    var profiles: Map<String, StructureDefinition> = HashMap()
)
